### DISCUSSION TITLE
Can we use Messagings insights toggle button pattern?

### DISCUSSION THREAD

**Discussion by mjanssen2021** [twilio-labs/paste#2626] [Q&A] [answered]
The Messaging > Insights page has these toggle button pattern (Image below highlighted in yellow), in which a user can toggle different ways to display the information in that card. 

Can we get approval from the design systems team to replicate this pattern ourselves in our Verify > SMS Fraud Insights dashboard data visualization page we are currently building (this is brand new functionality)? 

Or is there something we should keep in mind when replicating this pattern ourselves using the paste Toggle Button (https://paste.twilio.design/components/button#toggle-button) and high charts? Thanks. 

I attached our new concept mock below the older messaging one. 

<img width="1001" alt="Screen Shot 2022-08-22 at 8 19 28 AM" src="https://user-images.githubusercontent.com/78432794/185965302-409fd7c5-a5c1-4ce0-a9eb-0b427b347196.png">

<img width="948" alt="VerifySMSFraud_Mock" src="https://user-images.githubusercontent.com/78432794/185967805-21206515-f933-4ed0-a521-8db9ec2decf9.png">

**Comment by shleewhite** [comment, marked as answer]
Hi @mjanssen2021!

Can you clarify what you mean by replicating the pattern? Are you able to use the Paste Button or are you re-building it?

One note, the UX Infrastructure doesn't approve or sign off on designs at Twilio because we don't necessarily have all the product context needed to tell you if your designs are correct.

**Reply by mjanssen2021** [reply]
Hi Lee, ah good to know. We want to rebuild it using pate - but now knowing that ux infastructure doesn't necessarily sign off on these types of questions - would this be a better question to bring to a desgin systems office hours instead?

Thank you!

**Reply by shleewhite** [reply]
Sure! 

Here are a couple things to keep in mind when building a toggle button:
* Use the `aria-pressed` attribute to let all users know about the current state of the button. It is false when off and true when toggled on.
* The text of the button should not change with the state. So for your example, the bar chart button could have visually hidden text that says "Display as bar chart".
